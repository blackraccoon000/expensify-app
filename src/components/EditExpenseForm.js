import React from "react"
import moment from "moment"

export default class EditExpenseForm extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      amount: props.amount,
      calenderFocused: false,
      createdAt: moment(),
      description: props.description ,
      error:undefined,
      note: props.note
    }
  }
  onAmountChange = (e) => {
    const amount = e.target.value
    if(!amount || amount.match(/^\d{1,7}(\.\d{0,2})?$/)) {
      this.setState(()=>({ amount }))
    }
  }

  onDescriptionChange = (e) => {
    const description = e.target.value
    this.setState(()=>({ description }))
  }

  onNoteChange = (e) => {
    const note = e.target.value
    this.setState(()=>({ note }))
  }

  onSubmit = (e) => {
    e.preventDefault()
    if(!this.state.description || !this.state.amount) {
      this.setState(()=>{ error:"Please provide description and amount."})
    } else {
      this.setState(()=>{ error:undefined })
      this.props.onSubmit({
        description:this.state.description,
        amount: parseFloat(this.state.amount,10),
        createdAt: this.state.createdAt.valueOf(),
        note:this.state.note
      })
    }
  }

  render() {
    return (
      <div>
        <form onSubmit={this.onSubmit}>
          <label>
            Des:
            <input type="text" value={this.state.description} onChange={this.onDescriptionChange}/>
          </label>
          <label>
            Amount:
            <input type="text" value={this.state.amount} onChange={this.onAmountChange}/>
          </label>
          <label>
            Note:
            <textarea value={this.state.note} onChange={this.onNoteChange}></textarea>
          </label>
          <button>Edit Expense</button>
        </form>
      </div>
    )
  }
}