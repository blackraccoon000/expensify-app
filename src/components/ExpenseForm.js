import React from "react"
import moment from "moment"

// import { connect } from "react-redux"
// import { addExpense } from "../actions/expenses"
import { SingleDatePicker } from "react-dates"

export default class ExpenseForm extends React.Component {
  constructor(props){
    super(props)
    this.state = {
      amount:props.expense ? props.expense.amount.toString() : "",
      calenderFocused: false,
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      description: props.expense ? props.expense.description : "",
      error:undefined,
      note: props.expense ? props.expense.note : "",
    }
  }

  onAmountChange = (e) => {
    const amount = e.target.value

    if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)){
      // console.log(amount)
      this.setState(()=>({ amount }))
    }
  }
  onDateChange = (createdAt) => {
    if(createdAt) {
      this.setState(()=>({ createdAt }))
    }
  }
  onDescriptionChange = (e) => {
    const description = e.target.value
    this.setState(()=>({ description }))
  }
  onFocusChange = ({ focused }) => {
    // console.log(focused)
    this.setState(()=>({ calenderFocused: focused }))
  }
  onNoteChange = (e) => {
    const note = e.target.value
    this.setState(()=>({ note }))
  }
  onSubmit = (e) => {
    e.preventDefault()
    if(!this.state.description || !this.state.amount) {
      this.setState(()=>({ error:"Please provide description and amount." }))
    } else {
      this.setState(()=>({ error:undefined }))
      this.props.onSubmit({
        description:this.state.description,
        amount: parseFloat(this.state.amount,10),
        createdAt: this.state.createdAt.valueOf(),
        note:this.state.note
      })
    }
  }
  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <form onSubmit={this.onSubmit}>
          <input
            type="text" autoFocus placeholder="Description"
            value={this.state.description}
            onChange={this.onDescriptionChange}
          />
          <input
            type="text" placeholder="Amount"
            value={this.state.amount}
            onChange={this.onAmountChange}/>
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calenderFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={_=>false}
          />
          <textarea
            placeholder="Add a note for your expense (option)"
            value={this.state.note}
            onChange={this.onNoteChange}
          ></textarea>
          <button>Add Expense</button>
        </form>
        {this.state.error && <p>{this.state.error}</p>}
      </div>
    )
  }
}

// const ExpenseForm = ({dispatch}) => (
//   <div>
//     <form
//       onSubmit={e=>{
//         e.preventDefault()
//         dispatch(addExpense({
//           description:e.target.elements[0].value,
//           amount:e.target.elements[1].value
//           }))
//         console.log("created:",e.target.elements[0].value)
//         e.target.elements[0].value = ""
//         e.target.elements[1].value = ""
//         }
//       }
//     >
//       <label>Des:
//         <input type="text" placeholder="Description" autoFocus name="Des"/>
//         <input type="text" placeholder="Amount" autoFocus name="Amo"/>
//       </label>
//         <input type="submit" value="submit"/>
//     </form>
//   </div>
// )

// export default connect()(ExpenseForm)