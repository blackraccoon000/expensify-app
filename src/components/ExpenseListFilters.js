import React from "react"
import { connect } from "react-redux"
import { DateRangePicker } from "react-dates"
import { setTextFilter,sortByAmount,sortByDate, setStartDate,setEndDate } from "../actions/filters"

export class ExpenseListFilters extends React.Component {
  constructor(props){
    super(props)
    this.state = {
      calendarFocused:null
    }
  }
  onDatesChange = ({startDate, endDate}) => {
    this.props.startDate(startDate)
    this.props.endDate(endDate)
  }
  onFocusChange = ( calendarFocused ) => {
    this.setState({ calendarFocused })
  }
  onSetTextFilter = (e) => {
    this.props.setTextFilter(e.target.value)
  }
  onSortChange = (e) => {
    if(e.target.value==="date") {
      this.props.sortByDate()
    } else if (e.target.value==="amount") {
      this.props.sortByAmount()
    }
  }
  render() {
    const {filters} = this.props
    return (
      <div>
      <input type="text" value={filters.text}
        onChange = { this.onSetTextFilter }
      />
      <select
        value={filters.sortBy}
        onChange={this.onSortChange}>
        <option value="date">Date</option>
        <option value="amount">Amount</option>
      </select>
      <DateRangePicker
        startDate={filters.startDate}
        endDate={filters.endDate}
        onDatesChange={this.onDatesChange}
        focusedInput={this.state.calendarFocused}
        onFocusChange={this.onFocusChange}
        numberOfMonths={1} // 1月ごと表示
        isOutsideRange={()=>false} // 範囲外表示
        showClearDates={true} //　☓ボタン
      />
    </div>
    )
  }
}

const mapStateToProps = (state) => ({
  filters: state.filters
})

const mapDispatchToProps = (dispatch) => ({
  startDate: startDate => dispatch(setStartDate(startDate)),
  endDate: endDate => dispatch(setEndDate(endDate)),
  setTextFilter: text => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount())
})

export default connect(mapStateToProps,mapDispatchToProps)(ExpenseListFilters)