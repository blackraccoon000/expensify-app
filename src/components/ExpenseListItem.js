import React from "react"
import { Link } from "react-router-dom"
import moment from "moment"
import numeral from "numeral"

const ExpenseListItem = ({id,description,amount,createdAt}) => {
  return (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>Des:{description}</h3>
    </Link>
    <p>
      amount:{numeral(amount / 100).format("¥0,0.00")}
      -
      createdAt:{moment(createdAt).format("YYYY[/]MM[/]DD , HH:mm:ss")}
    </p>
  </div>
)}

// export default connect()(ExpenseListItem)
export default　ExpenseListItem