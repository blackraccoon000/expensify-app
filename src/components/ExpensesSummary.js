import React from "react"
import { connect } from "react-redux"
import { Link } from "react-router-dom"
import selectExpenses from "../selectors/expenses"
import selectExpensesTotal from "../selectors/expenses-total.js"
import numeral from "numeral"

class ExpensesSummary extends React.Component {
  constructor(props) {
    super(props)
  }
  render() {
    // Count
    const expenseCount = this.props.expenseCount
    const allExpenseCount = this.props.allExpenseCount
    const hiddenExpensesCount = allExpenseCount - expenseCount

    // Word
    const expenseWord = expenseCount === 1 ? "expense" : "expenses"
    const allExpenseWord = allExpenseCount === 1 ? "expense" : "expenses"
    const hiddenExpensesCountWord = hiddenExpensesCount === 1 ? "expense" : "expenses"

    // Total
    const expensesTotal = this.props.expensesTotal
    const allExpensesTotal = this.props.allExpensesTotal
    const hiddenExpensesTotal = allExpensesTotal - expensesTotal

    // Format
    const formattedExpensesTotal = numeral(expensesTotal / 100).format("$0,0.00")
    const formattedAllExpensesTotal = numeral(allExpensesTotal / 100).format("$0,0.00")
    const formattedHiddenExpensesTotal = numeral(hiddenExpensesTotal / 100).format("$0,0.00")

    // Message
    const viewingMessage = <h1 className="page-header__title">Viewing <span>{this.props.expenseCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h1>
    const allMessage = <h2 className="page-header__subTitle">All {allExpenseWord} <span>{allExpenseCount}</span> {allExpenseWord} totalling <span>{formattedAllExpensesTotal}</span></h2>
    const hiddenMessage = <h2 className="page-header__subTitle">Hidden {hiddenExpensesCountWord} <span>{hiddenExpensesCount}</span> {hiddenExpensesCountWord} totalling <span>{formattedHiddenExpensesTotal}</span> (All {formattedAllExpensesTotal})</h2>

    return (
      <div className="page-header">
        <div className="content-container">
          {expenseCount ? viewingMessage : <h2 className="page-header__alert">No Viewing expenses.</h2>}
          {hiddenExpensesCount ?  hiddenMessage : undefined}

          <div className="page-header__actions">
            <Link className="button" to="/create">Add Expense</Link>
          </div>
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)
  const allExpenses = selectExpenses(state.expenses)
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses),
    allExpenseCount: allExpenses.length,
    allExpensesTotal: selectExpensesTotal(allExpenses)
  }
}

export default connect(mapStateToProps)(ExpensesSummary)
export { ExpensesSummary }