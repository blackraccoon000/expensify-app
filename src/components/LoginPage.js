import React from "react"
import { connect } from "react-redux"
import { startLogin, twitterLogin, facebookLogin } from "../actions/auth"
class LoginPage extends React.Component {
  constructor(props){
    super(props)
  }
  render() {
    return (
      <div className="box-layout">
        <div className="box-layout__box">
          <h1 className="box-layout__title">Expensify App</h1>
          <p className="box-layout__subTitle">It's time to get your expenses under control.</p>
          <div className="box-layout__buttonArea">
            <button className="google button button--login" onClick={this.props.startLoginDispatcher}>
              <i className="fi-social-google-plus"></i> Google
            </button>
            <button className="twitter button button--login" onClick={this.props.twitterLoginDispatcher}>
              <i className="fi-social-twitter"></i> Twitter
            </button>
            <button className="facebook button button--login" onClick={this.props.facebookLoginDispatcher}>
              <i className="fi-social-facebook"></i> Facebook
            </button>
          </div>
        </div>
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  startLoginDispatcher: () => dispatch(startLogin()),
  twitterLoginDispatcher: () => dispatch(twitterLogin()),
  facebookLoginDispatcher: () => dispatch(facebookLogin())
})

export { LoginPage }
export default connect( undefined, mapDispatchToProps )(LoginPage)