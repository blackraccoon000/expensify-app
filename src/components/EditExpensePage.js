import React from "react"
import { connect } from "react-redux"
import { editExpense,startRemoveExpense } from "../actions/expenses"
import ExpenseForm from "./ExpenseForm"

export class EditExpensePage extends React.Component {
  constructor(props) {
    super(props)
  }
  onRemove= _ =>{
    this.props.startRemoveExpense({id:this.props.expense.id})
    this.props.history.push("/")
  }
  onSubmit= updates =>{
    this.props.editExpense(this.props.expense.id,updates)
    this.props.history.push("/")
  }
  render() {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />
        <button
          onClick={this.onRemove}
        >remove</button>
      </div>
    )
  }
}

const mapStateToProps = (state,props) => ({
  expense: state.expenses.find(expense => expense.id === props.match.params.id)
})

const mapDispatchToProps = (dispatch,props) => ({
  editExpense: (id,updates) => dispatch(editExpense(id,updates)),
  startRemoveExpense: id => dispatch(startRemoveExpense(id))
})

export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage)