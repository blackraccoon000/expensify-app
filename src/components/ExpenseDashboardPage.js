import React from "react"
import ExpenseList from "./ExpenseList"
import ExpenseListFilters from "./ExpenseListFilters"

const ExpenseDashboardPage = () => {
  return (
  <div>
    This is from my dashboard component
    <ExpenseListFilters />
    <ExpenseList />
  </div>
)}

export default ExpenseDashboardPage