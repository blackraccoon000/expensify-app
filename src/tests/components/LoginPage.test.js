import React from "react"
import { LoginPage } from "../../components/LoginPage"
import { shallow } from "enzyme"

let wrapper, startLoginDispatcher

beforeEach(()=>{
  startLoginDispatcher = jest.fn()
  wrapper = shallow(<LoginPage startLoginDispatcher={startLoginDispatcher}/>)
})

it("should render LoginPage Component", () => {
  expect(wrapper).toMatchSnapshot()
})

it("should call startLogin on button click", () => {
  wrapper.find("button").simulate("click")
  expect(startLoginDispatcher).toHaveBeenCalled()
})