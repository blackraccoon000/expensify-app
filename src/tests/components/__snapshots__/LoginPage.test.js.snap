// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage Component 1`] = `
<div>
  <p>
    Welcome Login Page
  </p>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
